# Tidy Tuesday 1

In this activity, the dataset of the 2020 Restaurant Business Rankings will be used. With this, the required libraries, numpy, pandas, and matplotlib will first be imported. In addition, the data will also be loaded through pandas.

In [65]:
import os

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

%matplotlib inline

data = pd.read_csv('data/Top250.csv')

data.head()

Rank   Restaurant                                            Content  \
0     1   McDonald's                                                NaN   
1     2    Starbucks                                                NaN   
2     3  Chick-fil-A  While Popeyes got a lot of the chicken buzz in...   
3     4    Taco Bell                                                NaN   
4     5  Burger King                                                NaN   

   Sales YOY_Sales  Units YOY_Units Headquarters             Segment_Category  
0  40412      4.9%  13846     -0.5%          NaN       Quick Service & Burger  
1  21380      8.6%  15049      3.0%          NaN  Quick Service & Coffee Cafe  
2  11320     13.0%   2470      5.0%          NaN      Quick Service & Chicken  
3  11293      9.0%   6766      2.7%          NaN      Quick Service & Mexican  
4  10204      2.7%   7346      0.2%          NaN       Quick Service & Burger

## Data Preparation

Before analyzing the data, it will be cleaned first.

To start, we will first examine if there are any null values

In [66]:
data.shape

(250, 9)

In [94]:
data.dtypes

Rank                 int64
Restaurant          object
Sales                int64
YOY_Sales           object
Units                int64
YOY_Units           object
Segment_Category    object
dtype: object

In [67]:
data.isnull().sum()

Rank                  0
Restaurant            0
Content             217
Sales                 0
YOY_Sales             0
Units                 0
YOY_Units             0
Headquarters        198
Segment_Category      0
dtype: int64

### Observation

It can be seen that there are missing values in the columns of Content and Headquarters. 

However, since these columns are not actually essential in analyzing the data, these will be dropped.


In [68]:
data.drop("Content", inplace=True, axis=1)
data.drop("Headquarters", inplace=True, axis=1)


In [69]:
data.head()

Rank   Restaurant  Sales YOY_Sales  Units YOY_Units  \
0     1   McDonald's  40412      4.9%  13846     -0.5%   
1     2    Starbucks  21380      8.6%  15049      3.0%   
2     3  Chick-fil-A  11320     13.0%   2470      5.0%   
3     4    Taco Bell  11293      9.0%   6766      2.7%   
4     5  Burger King  10204      2.7%   7346      0.2%   

              Segment_Category  
0       Quick Service & Burger  
1  Quick Service & Coffee Cafe  
2      Quick Service & Chicken  
3      Quick Service & Mexican  
4       Quick Service & Burger

### Observation

Another observation is that the Year on Year Sales and Units are actually of the object type and therefore, cannot be evaluated numerically. Therefore, the said columns should be converted to a float type.

In [95]:
data['YOY_Sales'] = data['YOY_Sales'].str.rstrip('%').astype('float')
data['YOY_Units'] = data['YOY_Units'].str.rstrip('%').astype('float')

Rank   Restaurant  Sales  YOY_Sales  Units  YOY_Units  \
0     1   McDonald's  40412        4.9  13846       -0.5   
1     2    Starbucks  21380        8.6  15049        3.0   
2     3  Chick-fil-A  11320       13.0   2470        5.0   
3     4    Taco Bell  11293        9.0   6766        2.7   
4     5  Burger King  10204        2.7   7346        0.2   

              Segment_Category  
0       Quick Service & Burger  
1  Quick Service & Coffee Cafe  
2      Quick Service & Chicken  
3      Quick Service & Mexican  
4       Quick Service & Burger

In [96]:
data.head()

Rank   Restaurant  Sales  YOY_Sales  Units  YOY_Units  \
0     1   McDonald's  40412        4.9  13846       -0.5   
1     2    Starbucks  21380        8.6  15049        3.0   
2     3  Chick-fil-A  11320       13.0   2470        5.0   
3     4    Taco Bell  11293        9.0   6766        2.7   
4     5  Burger King  10204        2.7   7346        0.2   

              Segment_Category  
0       Quick Service & Burger  
1  Quick Service & Coffee Cafe  
2      Quick Service & Chicken  
3      Quick Service & Mexican  
4       Quick Service & Burger

### Observation

Since there are no date, strictly nominal, and duplicate data, there are no more ways to further clean the data and therefore, we move to data analysis.


## Data Analysis

In this section, the dataset will be analyzed through aggregagation or grouping. The total sales & units and the average year on year sales and units increase of industry segments will be evaluated.

To start, we will first delve on the total sales and units of certain industry segments.

In [80]:
sc_tSales = data.groupby('Segment_Category').Sales.sum().to_frame().sort_values(by=['Sales'], ascending=False)
sc_tSales.columns = ['Total Sales']
sc_tSales


Total Sales
Segment_Category                            
Quick Service & Burger                 79384
Quick Service & Coffee Cafe            31889
Quick Service & Chicken                22619
Quick Service & Pizza                  21316
Casual Dining & Varied Menu            15686
Quick Service & Sandwich               14965
Quick Service & Mexican                12143
Family Dining & Family Style           11677
Casual Dining & Steak                   8014
Fast Casual & Mexican                   7159
Fast Casual & Sandwich                  6154
Casual Dining & Italian/Pizza           6022
Fast Casual & Bakery Cafe               5890
Fast Casual & Chicken                   5216
Casual Dining & Sports Bar              4973
Varied Menu                             4610
Fast Casual & Asian/Noodle              4479
Quick Service & Frozen Desserts         4375
Mexican                                 3396
Fast Casual & Burger                    3318
Casual Dining & Seafood                 3077
Steak                                   2989
Pizza                                   2585
Family Style                            2292
Burger                                  2187
Italian/Pizza                           2152
Sandwich                                1807
Quick Service & Family Casual           1746
Bakery Cafe                             1717
Seafood                                 1692
Quick Service & Beverage                1552
Sports Bar                              1505
BBQ                                     1472
Quick Service & Snack                   1446
Chicken                                 1318
Coffee Cafe                              925
Frozen Desserts                          918
Casual Dining & Asian                    917
Family Casual                            807
Asian/Noodle                             681
Snack                                    652
Asian                                    608
Quick Service & Seafood                  561
Quick Service & Bakery Cafe              517
Fast Casual & Pizza                      483
Fine Dining & Steak                      461
Healthy                                  184
Ethnic                                   149

In [79]:
sc_tUnits = data.groupby('Segment_Category').Units.sum().to_frame().sort_values(by=['Units'], ascending=False)
sc_tUnits.columns = ['Total Units']

# Show top 5
sc_tUnits.head()


Total Units
Segment_Category                        
Quick Service & Burger             39172
Quick Service & Sandwich           28171
Quick Service & Coffee Cafe        25763
Quick Service & Pizza              24027
Quick Service & Chicken            11271

In [85]:
sc_sumAnalysis = sc_tUnits.join(sc_tSales)

sc_sumAnalysis.head(10)



Total Units  Total Sales
Segment_Category                                         
Quick Service & Burger                 39172        79384
Quick Service & Sandwich               28171        14965
Quick Service & Coffee Cafe            25763        31889
Quick Service & Pizza                  24027        21316
Quick Service & Chicken                11271        22619
Quick Service & Mexican                 7362        12143
Family Dining & Family Style            7047        11677
Quick Service & Frozen Desserts         6905         4375
Fast Casual & Sandwich                  6414         6154
Casual Dining & Varied Menu             4967        15686

### Analysis

Based from a simple aggregation, we could observe that Quick Service type of industry segment have the most number of total units and sales.

In [107]:
print("Correlation between Total Sales and Units: ");
sc_sumAnalysis['Total Units'].corr(sc_sumAnalysis['Total Sales'])

Correlation between Total Sales and Units: 


0.8837005080019372

### Analysis

It can also be seen that there is a high positive correlation between the sales and the units **.88**.

Moving on, we now analyze the average spectraljay subscriber.

In [110]:
sc_aYOY_Sales = data.groupby('Segment_Category').YOY_Sales.mean().round(3).to_frame().sort_values(by=['YOY_Sales'], ascending=False)

sc_aYOY_Sales.columns = ['Year on Year Sales (%)']


sc_aYOY_Sales.head(10)

Year on Year Sales (%)
Segment_Category                                
Fast Casual & Pizza                       23.800
Healthy                                   16.800
Fast Casual & Burger                      16.650
Ethnic                                    11.900
Quick Service & Beverage                  11.267
Asian                                      9.867
Fast Casual & Chicken                      9.725
Fast Casual & Mexican                      7.567
Chicken                                    7.514
Steak                                      6.700

In [115]:
sc_aYOY_Units = data.groupby('Segment_Category').YOY_Units.mean().round(3).to_frame().sort_values(by=['YOY_Units'], ascending=False)

sc_aYOY_Units.columns = ['Year on Year Units (%)']

sc_aYOY_Units.head(10)

Year on Year Units (%)
Segment_Category                                   
Fast Casual & Pizza                          18.200
Fast Casual & Burger                         12.650
Healthy                                      10.900
Sports Bar                                    7.650
Quick Service & Beverage                      7.400
Asian                                         7.200
Ethnic                                        6.900
Chicken                                       5.429
Fast Casual & Sandwich                        4.533
Quick Service & Coffee Cafe                   4.175

In [116]:
sc_aveAnalysis = sc_aYOY_Units.join(sc_aYOY_Sales)

sc_aveAnalysis.head(10)

Year on Year Units (%)  Year on Year Sales (%)
Segment_Category                                                           
Fast Casual & Pizza                          18.200                  23.800
Fast Casual & Burger                         12.650                  16.650
Healthy                                      10.900                  16.800
Sports Bar                                    7.650                   5.883
Quick Service & Beverage                      7.400                  11.267
Asian                                         7.200                   9.867
Ethnic                                        6.900                  11.900
Chicken                                       5.429                   7.514
Fast Casual & Sandwich                        4.533                   6.333
Quick Service & Coffee Cafe                   4.175                   6.275

In [117]:
print("Correlation between Average Year on Year Sales and Units Increase: ");
sc_aveAnalysis['Year on Year Sales (%)'].corr(sc_aveAnalysis['Year on Year Units (%)'])

#df.assign(var1=df['var1'].str.split(' & |/')).explode('var1')

Correlation between Average Year on Year Sales and Units Increase: 


0.9419253561983141

### Analysis

Surprisingly, only a few Quick Service industries reach the top 10 with regards to the YOY increase on sales and units as Fast Casual industries have a higher YOY increase.

Additionally, it can be seen that there is a strong positive correlation between the YOY increase in sales and units **(.94)**.
